### PR TITLE
Worksheet improvements

### PR DESCRIPTION
**Pull request by smarter**

### CODE REVIEW

File: vscode-dotty/src/features.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    ensure(ensure(capabilities, "worksheet")!, "exec")!.dynamicRegistration = true
+  }
+
+  public initialize(capabilities: ServerCapabilities & WorksheetServerCapabilities, documentSelector: DocumentSelector): void {
+    if (!capabilities.worksheetRunProvider) {
+      return
+    }
+
+    const selector: DocumentSelector = documentSelector || [ { language: 'scala', pattern: '**/*.sc' } ]
+    if (selector) {
```
Comment (Duhemm) [inline vscode-dotty/src/features.ts]: Why is that condition needed? `selector` can never be `nil`, no?

File: vscode-dotty/src/features.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+export class WorksheetRunFeature extends TextDocumentFeature<TextDocumentRegistrationOptions> {
+	constructor(client: BaseLanguageClient) {
+		super(client, WorksheetRunRequest.type)
+	}
+
+  public fillClientCapabilities(capabilities: ClientCapabilities & WorksheetClientCapabilities): void {
+    ensure(ensure(capabilities, "worksheet")!, "run")!.dynamicRegistration = true
+  }
```
Comment (Duhemm) [inline vscode-dotty/src/features.ts]: `ensure` adds the specified keys to the given object is they're missing, is that it?

Comment (smarter) [reply vscode-dotty/src/features.ts]: Yes, I think so, I adapted this from what's done in other features `fillClientCapabilities` in vscode-languageclient

**Comment by smarter** [comment]
Going to merge this as-is and open another PR for further improvements.

**Pull request merged**